src/bsoncxx/include/bsoncxx/v_noabi/bsoncxx/builder/concatenate.hpp

```diff
@@ -26,8 +26,10 @@ namespace v_noabi {
 namespace builder {
 
 ///
-/// Container to concatenate a document. Use it by constructing an instance with the
-/// document to be concatenated, and pass it into a document stream builder.
+/// Container to concatenate a document.
+///
+/// Use this with a document or array builder to merge an existing document's fields with that of
+/// the document or array being built.
```

Comment:
```suggestion
/// Use this with a document builder to merge an existing document's fields with that of the
/// document being built.
```

I expect the concatenate only works with array+array or document+document:

```c++
using bsoncxx::builder::concatenate;
using bsoncxx::builder::basic::kvp;
using bsoncxx::builder::basic::make_array;
using bsoncxx::builder::basic::make_document;

auto arr = make_array(1, 2, 3);
auto doc = make_document(kvp("foo", "bar"));

// Cannot concatenate an array to a document builder
{
    bsoncxx::builder::basic::document builder;
    builder.append(concatenate(doc.view()));     // OK
    // builder.append(concatenate(arr.view()));  // Error: no matching member function
}

// Cannot concatenate a document to an array builder
{
    bsoncxx::builder::basic::array builder;
    // builder.append(concatenate(doc.view()));  // Error: no matching member function
    builder.append(concatenate(arr.view()));     // OK
}
```

Similar suggestion applies to other concatenate helpers.